dot/network/status_test.go

```diff
@@ -109,5 +113,141 @@ func TestStatus(t *testing.T) {
 	}
 }
 
-//have a peer send a message status with a block ahead
-// check that the message returned is the expected
+// helper method to create and start a new network service
+func createTestServiceWithBlockState(t *testing.T, cfg *Config, blockState *MockBlockState) (node *Service, msgSend chan Message, msgRec chan Message) {
+	msgRec = make(chan Message)
+	msgSend = make(chan Message)
+
+	// same for all network tests use the createTestService helper method
+	cfg.BlockState = blockState // required
+	cfg.NetworkState = &MockNetworkState{}
+	cfg.ProtocolID = TestProtocolID // default "/gossamer/dot/0"
+
+	node, err := NewService(cfg, msgSend, msgRec)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = node.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return node, msgSend, msgRec
+}
+
+// have a peer send a message status with a block ahead
+// test exchanged messages after peer connected are correct
+func TestStatusAndBlockRequestMessage(t *testing.T) {
+	dataDirA := path.Join(os.TempDir(), "gossamer-test", "nodeA")
+	defer os.RemoveAll(dataDirA)
+
+	configA := &Config{
+		DataDir:     dataDirA,
+		Port:        7001,
+		RandSeed:    1,
+		NoBootstrap: true,
+		NoMdns:      true,
+	}
+
+	blockState := newMockBlockState(big.NewInt(3))
```

Comment:
can we use `blockStateA` and `blockStateB` in this test to be consistent and clear we are not using the same state?

Reply:
yes